File: src/tasks/cli/feature/start/StartParser.java
package tasks.cli.feature.start;

import omnia.data.cache.Memoized;
import omnia.data.structure.List;
import tasks.cli.command.common.simple.SimpleParser;
import tasks.cli.parser.ParseResult;
import tasks.cli.parser.Parser;
import tasks.model.Task;

/** Command line argument parser for the Start command. */
public final class StartParser extends SimpleParser<StartArguments> {
  public StartParser(
      Memoized<? extends Parser<? extends List<? extends ParseResult<? extends Task>>>>
          taskParser) {
    super(StartArguments::new, taskParser);
  }
}

File: cli/src/me/strangepan/tasks/cli/feature/start/StartParser.kt
package me.strangepan.tasks.cli.feature.start

import java.util.function.Function
import omnia.data.cache.Memoized
import omnia.data.structure.List
import me.strangepan.tasks.cli.command.common.simple.SimpleParser
import me.strangepan.tasks.cli.parser.ParseResult
import me.strangepan.tasks.cli.parser.Parser
import me.strangepan.tasks.engine.model.Task

/** Command line argument parser for the Start command.  */
class StartParser(
    taskParser: Memoized<out Parser<out List<out ParseResult<out Task>>>>) : SimpleParser<StartArguments>(Function { tasks: List<Task> -> StartArguments(tasks) }, taskParser)